nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -27,16 +26,38 @@ def __init__(self, **source_data):
 
     def get_metadata_schema(self):
         """Compile metadata schema for the RecordingExtractor."""
-        metadata_schema = get_base_schema(
-            properties=dict(SpikeEventSeries=get_schema_from_hdmf_class(SpikeEventSeries))
+        metadata_schema = super().get_metadata_schema()
+
+        # Initiate Ecephys metadata
+        metadata_schema["properties"]["Ecephys"] = get_base_schema(tag="Ecephys")
+        metadata_schema["properties"]["Ecephys"]["required"] = []
+        metadata_schema["properties"]["Ecephys"]["properties"] = dict(
+            UnitProperties=dict(
+                type="array",
+                minItems=0,
+                renderForm=False,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/UnitProperties"},
```

Comment:
I think it's a bit odd to have definitions deep in the schema as opposed to just at the root level

Reply:
Yet another question for @luiztauffer, I don't know why this is the way it is for the forms. Any ideas or simpler ways to do it @luiztauffer?

Reply:
@luiztauffer why are we using definitions here

Reply:
@bendichter @CodyCBakerPhD 
tbh I don’t remember why we made this choice, it was a while back. 
Since these schemas might be overwritten by subsequent DataInterface schemas in a multi-interface Converter, maybe it makes sense to keep this level of separation by modules, to avoid eventual overwriting of same definitions but with different default values, for example. But I can’t really think on an example where this would happen right now.
In principle these should work just the same, when it comes to validation:
- {"$ref": "#/properties/Ecephys/properties/definitions/UnitProperties"}
- {"$ref": "#/properties/definitions/UnitProperties"}
- {"$ref": "#/definitions/UnitProperties"}

But I can’t guarantee changing this won’t break the converters or the GUI Forms at this moment.
We can open an Issue for that, if you think these definitions should move to the root.